newIDE/app/src/InstructionOrExpression/EnumerateInstructions.js

```diff
@@ -51,13 +52,24 @@ const isObjectInstruction = (
   ) {
     return false;
   }
-  if (firstParameterIndex === instructionMetadata.getParametersCount() - 1) {
+
+  if (!objectBehaviorTypes) {
     return true;
   }
-  const secondParameter = instructionMetadata.getParameter(
-    firstParameterIndex + 1
-  );
-  return !gd.ParameterMetadata.isBehavior(secondParameter.getType());
+  for (
+    let parameterIndex = firstParameterIndex + 1;
+    parameterIndex < instructionMetadata.getParametersCount();
+    parameterIndex++
+  ) {
+    const parameter = instructionMetadata.getParameter(parameterIndex);
+    if (!gd.ParameterMetadata.isBehavior(parameter.getType())) {
+      return true;
```

Comment:
I'm having trouble knowing if this is correct: the function is named `isObjectInstruction`, but you return true if the parameter is not a behavior (or linked to?). I'm not sure about that. Some comments might be helpful

Reply:
I added comments.